server/msteams/client.go

```diff
@@ -140,14 +141,20 @@ func NormalizeGraphAPIError(err error) error {
 				message = *terr.GetMessage()
 			}
 			return &GraphAPIError{
-				Code:    code,
-				Message: message,
+				Code:       code,
+				Message:    message,
+				StatusCode: e.ResponseStatusCode,
```

Comment:
I'm getting the status code to detect the 401 cases that means the token is not valid.

Reply:
If this is true:

> The 401 (Unauthorized) status code indicates that the request has not been applied because it lacks valid authentication credentials for the target resource.

Then we can't just trigger `OnDisconnect` simply because a particular resource is unavailable, as a reconnection wouldn't help anyway, and this might lead to a stream of "random" disconnects.

Can we constrain our disconnection logic more narrowly?

Reply:
We can, but take into account that 401 (Unathorized) means actually, "Unathenticated", so... the token that you are using is not valid. Is not "you don't have permission", that would be 403. We can avoid this approach for now, and just observe it for now, if we see a lot of 401 happening, then, that would lead to random disconnections, if there are no 401, or they are anetdotical, that means that this is a valid approach. What do you think?